### PR TITLE
[CI] Added GraphQL API Documentation Workflow

### LINKED ISSUE #3077: [CI] GraphQL API Documentation Workflow
#### Desired Behavior
Create a CI/CD workflow for automatic Document generation of `GraphQL APIs`
#### Implementation

- The workflow needs to be triggered when `internal/graphql/schema/schema.graphql` is changed.
- The workflow needs to run `bundle install ; bundle exec rake graphql:compile_docs` inside docs folder to updates the `docs/pages/extensibility/graphql-reference.md`.
- Create nessecory `make` commands.

#### Acceptance Tests

- In case of Invalid schema, the workflow should fail.
 Eg. More than one input in a mutation
  ```graphQL
  type Mutation {
	  # This mutation is invalid
	  changeAddonStatus(inputA: String, inputB: String ): String!
  }
  ``` 
#### Reference:
- [Meshery Build & Release Strategy](https://docs.google.com/document/d/11nAxYtz2SUusCYZ0JeNRrOLIxkgmmbUVWz63MBZV2oE/edit#)

### PR DESCRIPTION
**Pull request by aminoxix**
_Signed-off-by: @amino19 <anshumaankrprasad76@gmail.com>_

**Description**
Added GraphQL API Documentation Workflow: with Ruby

This PR fixes #3077 

**Notes for Reviewers**
Needs review! 
Thanks to @DelusionalOptimist, @ramrodo & @dhruv0000 Sir. :slightly_smiling_face: 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by ramrodo** [comment]
@amino19 the current error is:

```
make: *** No rule to make target 'build-docs'.  Stop.
```

So, that is because in the Makefile there is no `build-docs` target. Not sure if you want to add it or you want use an existing Makefile target.

**Comment by warunicorn19** [comment]
Hey @amino19 , you're using build docs again, that's why you're running the these issues.
If you want to add it you'll have to make changes to the Makefile.

**Comment by amino19** [comment]
@ramrodo  & @warunicorn19 : Sir, actually I have to make workflow according to <a href="https://github.com/meshery/meshery/blob/master/Makefile#L168">this</a>, as suggested by @dhruv0000 sir. :slightly_smiling_face:

**Comment by dhruv0000** [comment]
@amino19 This commands should work.

```
    - name: graphql-docs
      run: |
        cd docs
        bundle install
        bundle exec rake graphql:compile_docs
```

**Comment by amino19** [comment]
> @amino19 This commands should work.
> 
> ```
>     - name: graphql-docs
>       run: |
>         cd docs
>         bundle install
>         bundle exec rake graphql:compile_docs
> ```

@dhruv0000 : Changes had been made, including commands & deleting strategy. Needs review! :slightly_smiling_face:

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:2]:
```suggestion
on:
  push:
    branches:
      - 'master'
    paths:
      - 'internal/graphql/schema/schema.graphql'   
```
@amino19 Let's modify this a bit so that it triggers only when the graphql schema is changed.
Something similar on pull_request events as well.

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:7]: This should checkout appropriate branch with right access tokens. For e.g look https://github.com/meshery/meshery/blob/master/.github/workflows/error-codes-updater.yaml

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:26]: We'll also need to make sure that this here commits to master

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]
Going great @amino19. Some minor changes. Let's get them done and then we'll look at further enhancements :smile:

File: .github/workflows/graphql.yml
Comment (leecalcote) [reply .github/workflows/graphql.yml:2]: Some great feedback happening in this review.

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:40]:
```suggestion
        commit_message: "[Docs] Generated documentation for GraphQL API"
        branch: master
```

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:33]:
```suggestion
      uses: stefanzweifel/git-auto-commit-action@v4
      if: github.event_name != 'pull_request'
```
This here will ensure that we don't make commits to master when this test runs in PRs

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:15]: @amino19 Will you also be able to add a condition here so that this job gets triggered only when any of these activities happen on `'meshery/meshery'` repo? :slightly_smiling_face:

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:7]: No @amino19 I wasn't talking about changing this. :sweat_smile:

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:15]: I've updated the suggestion here

File: .github/workflows/graphql.yml
Comment (amino19) [reply .github/workflows/graphql.yml:7]: @DelusionalOptimist : Okay only `pull_request`, must be updated by `meshery/meshery`?

Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:7]: @amino19 both `push` and `pull_request` events should be triggered only when `internal/graphql/schema/schema.graphql` changes.
That is, it was good the way it was earlier : )

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:15]: @amino19 This is where you'll be placing the condition on `meshery/meshery`

Comment (amino19) [reply .github/workflows/graphql.yml:15]: @DelusionalOptimist : Got it, Thanks! :slightly_smiling_face:

File: .github/workflows/graphql.yml
Comment (amino19) [reply .github/workflows/graphql.yml:41]: @DelusionalOptimist : Sir, Anything more after `branch: master`? :>

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:15]: Great :+1:

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:42]: No @amino19. Only adding this would do it :+1:

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:24]: @amino19 any reason for using this particular commit and not a stable release of the setup-ruby action?

Comment (amino19) [reply .github/workflows/graphql.yml:24]: @DelusionalOptimist : Are you concern about <a href= "https://docs.github.com/en/actions/guides/building-and-testing-ruby"> this</a>? Is it standard?

Comment (amino19) [reply .github/workflows/graphql.yml:24]: @DelusionalOptimist : Actually doesn't have any specific reason, just I saw this in GitHub marketplace while started writing workflow. :slightly_smiling_face: 
BTW, Are you concern about <a href= "https://docs.github.com/en/actions/guides/building-and-testing-ruby"> this</a>? Is it standard?

Comment (DelusionalOptimist) [reply .github/workflows/graphql.yml:24]:
```suggestion
      uses: ruby/setup-ruby@v1
```
I see. So let's use the v1 release to maintain consistency with other workflows. :+1:

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:16]: @amino19 this 32 sneaked in somehow :eyes:

**Comment by DelusionalOptimist** [comment]
@dhruv0000 does this look cool? @amino19 has made some changes so that:
* Runs the rake task and updates the docs whenever a commit is created on the master branch.
* Also runs the rake task in PRs to validate graphQL schema (doesn't update docs)

File: .github/workflows/graphql.yml
Comment (amino19) [reply .github/workflows/graphql.yml:16]: @DelusionalOptimist : My bad! :sweat_smile:

**Review by dhruv0000 — APPROVED** [review]

**Comment by dhruv0000** [comment]
@DelusionalOptimist Looks good. Waiting for your approval before Merging.

File: .github/workflows/graphql.yml
Comment (DelusionalOptimist) [inline .github/workflows/graphql.yml:21]:
```suggestion
```

**Comment by DelusionalOptimist** [comment]
Acceptance tests:
* https://github.com/luislopez3601/meshery/pull/14
* https://github.com/luislopez3601/meshery/commit/d2a4c24a70e74cc44ad501202612b553eb55f500

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. Thanks @amino19 :rocket:

**Pull request merged**
